pkg/route/secretmanager/manager.go

```diff
@@ -0,0 +1,150 @@
+package secretmanager
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/openshift/library-go/pkg/route/secret"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+type Manager struct {
+	// monitor for managing and watching "single" secret dynamically.
+	monitor secret.SecretMonitor
+
+	// Map of registered handlers for each route.
+	// Populated inside RegisterRoute() and used in UnregisterRoute(), GetSecret.
+	// generateKey() will create the map key.
+	registeredHandlers map[string]secret.SecretEventHandlerRegistration
+
+	lock sync.RWMutex
+
+	// Work queue to be used by the consumer of this Manager, mostly to add secret change events.
+	queue workqueue.RateLimitingInterface
+
+	// Event handler for secret changes.
+	secretHandler cache.ResourceEventHandler
+}
+
+func NewManager(kubeClient kubernetes.Interface, queue workqueue.RateLimitingInterface) *Manager {
+	return &Manager{
+		monitor:            secret.NewSecretMonitor(kubeClient),
+		lock:               sync.RWMutex{},
+		queue:              queue,
+		registeredHandlers: make(map[string]secret.SecretEventHandlerRegistration),
+		secretHandler:      nil,
```

Comment:
- Refer https://github.com/openshift/library-go/pull/1626#discussion_r1467005425
- During the initialization of `NewManager` we won’t have default (correct) secret handler, it’s necessary that the consumer should set it’s value with `WithSecretHandler` before calling `RegisterRoute`.
- NewManager can’t hold a definite secret handler, which is meant to change before every call.

Reply:
> NewManager can’t hold a definite secret handler, which is meant to change before every call.

Before every call of `RegisterRoute`? It doesn't make sense as  a field of `Manager` struct then, pass it at as an arg of `RegisterRoute`.

> During the initialization of NewManager we won’t have default (correct) secret handler, it’s necessary that the consumer should set it’s value with WithSecretHandler before calling RegisterRoute.

Do you have an example of the usage of `WithSecretHandler`?

Reply:
Yeah, I also think passing the `handler` as an arg of `RegisterRoute` would make more sense. I've refactored it, PTAL. 

> Do you have an example of the usage of WithSecretHandler?

Removed it in the latest changes, since we are not storing `handler` as a field of `Manager`